File: app/src/main/java/com/spendshare/dao/spendshare/MainActivity.java
package com.spendshare.dao.spendshare;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import com.spendshare.dao.spendshare.view.DiyDialog;

public class MainActivity extends AppCompatActivity {

    private Button mButton;
    private DiyDialog mDiyDialog;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initView();
    }

    private void initView() {
        mButton = (Button)findViewById(R.id.btn_show);
        mButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                mDiyDialog = new DiyDialog(MainActivity.this);
                mDiyDialog.setTitleStr("one");
                mDiyDialog.setMessageStr("towtwotwotwo");
                mDiyDialog.setYesOnclickListener(new DiyDialog.OnYesClickLisener() {
                    @Override
                    public void onYesClick() {
                        Toast.makeText(MainActivity.this,"点击了--确定--按钮",Toast.LENGTH_LONG).show();
                        mDiyDialog.dismiss();
                    }
                });
                mDiyDialog.setNoOnclickListener(new DiyDialog.OnNoClickListener() {
                    @Override
                    public void onNoClick() {
                        Toast.makeText(MainActivity.this,"点击了--取消--按钮",Toast.LENGTH_LONG).show();
                        mDiyDialog.dismiss();
                    }
                });
                mDiyDialog.show();
            }
        });
    }
}

File: app/src/main/java/com/spendshare/dao/spendshare/view/activity/MainActivity.kt
package com.spendshare.dao.spendshare.view.activity

import android.content.Intent
import android.os.Bundle
import android.support.design.widget.NavigationView
import android.support.design.widget.TabLayout
import android.support.v7.app.AppCompatActivity
import android.transition.Visibility
import android.view.Gravity
import android.view.MenuItem
import android.view.View
import android.widget.ImageView
import android.widget.Toast
import com.spendshare.dao.spendshare.R
import com.spendshare.dao.spendshare.R.id.action_check
import com.spendshare.dao.spendshare.R.id.action_task
import com.spendshare.dao.spendshare.model.MainModel
import com.spendshare.dao.spendshare.presenter.MainPresenter
import com.spendshare.dao.spendshare.view.fragment.CheckFragment
import com.spendshare.dao.spendshare.view.fragment.TaskAddFragmen
import com.spendshare.dao.spendshare.view.fragment.TaskFragment
import com.spendshare.dao.spendshare.view.iview.MainView
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(), MainView {

    private val presenter: MainPresenter = MainPresenter(this, MainModel())
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initView()
    }

    private fun initView() {
        navigation.setOnNavigationItemSelectedListener { menuItem ->
            when(menuItem.itemId){
                action_task -> {
                    supportFragmentManager.beginTransaction().replace(R.id.contentContainer,TaskFragment()).commit()
                    add_task.visibility = View.VISIBLE
                    true
                }
                action_check -> {
                    supportFragmentManager.beginTransaction().replace(R.id.contentContainer,CheckFragment()).commit()
                    true
                }
                else -> false
            }
        }
        navigation.selectedItemId = R.id.action_task
        val headerView = navigationView.getHeaderView(0).findViewById<ImageView>(R.id.drawerUserImage)
        headerView.setOnClickListener({view ->
            Toast.makeText(this@MainActivity, "hello", Toast.LENGTH_SHORT).show()
            drawerLayout.closeDrawers()
        })
        navigationView.setNavigationItemSelectedListener(NavigationView.OnNavigationItemSelectedListener { item: MenuItem ->
            Toast.makeText(this@MainActivity, item.title, Toast.LENGTH_SHORT).show()
            drawerLayout.closeDrawers()
            return@OnNavigationItemSelectedListener true
        })
        toolBar.navigationIcon = resources.getDrawable(R.drawable.ic_chevron_left)
        toolBar.setNavigationOnClickListener {
            supportFragmentManager.popBackStack()
        }
        add_task.setOnClickListener{
            supportFragmentManager.beginTransaction().replace(R.id.contentContainer,TaskAddFragmen()).commit()
            add_task.visibility = View.INVISIBLE
        }
    }

    override fun showDrawer() {
        drawerLayout.openDrawer(Gravity.LEFT)
    }

    override fun hideDrawer() {
        drawerLayout.closeDrawers()
    }
}
